slick.grid.js

```diff
@@ -5949,12 +5951,24 @@ if (typeof Slick === "undefined") {
       }
     }
 
-    /** basic html sanitizer to avoid scripting attack */
-    function sanitizeHtmlString(dirtyHtml) {
-      var sanitizer = options.sanitizer || function (dirtyHtmlStr) {
-        return dirtyHtmlStr.replace(/(\b)(on\S+)(\s*)=|javascript:([^>]*)[^>]*|(<\s*)(\/*)script([<>]*).*(<\s*)(\/*)script(>*)|(&lt;)(\/*)(script|script defer)(.*)(&gt;|&gt;">)/gi, '');
+   /** basic html sanitizer to avoid scripting attack */
+    function defaultSanitizeHtmlString(dirtyHtml) {
+       return dirtyHtmlStr.replace(/(\b)(on[a-z]+)(\s*)=|javascript:([^>]*)[^>]*|(<\s*)(\/*)script([<>]*).*(<\s*)(\/*)script(>*)|(&lt;)(\/*)(script|script defer)(.*)(&gt;|&gt;">)/gi, '');
+    }
+
+    function externalSanitizeHtmlString(dirtyHtml) {
+      if (typeof dirtyHtml !== 'string') return dirtyHtml;
+      var sanitizer = options.sanitizer || defaultSanitizeHtmlString;
+      return sanitizer(dirtyHtml);
+    }
+
+    function internalSanitizeHtmlString(dirtyHtml) {
```

Comment:
what's the difference between the internal/external? I would think that 1 type is enough to drive them all across the project